src/Auth/BaseOAuthAuthentication.cs

```diff
@@ -1,139 +1,47 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
 using System.Net.Http;
 using System.Net.Http.Headers;
-using System.Text;
 using System.Threading.Tasks;
 using IdentityModel.Client;
 
 namespace Jpp.Common.Backend.Auth
 {
     public abstract class BaseOAuthAuthentication : IAuthentication
     {
-        public bool Authenticated { get; private set; }
+        internal ErrorHandler _errorHandler;
+        protected IMessageProvider _messenger;
 
-        private string _accessToken;
-        private string _idToken;
-        private HttpClient _client;
-
-        private IMessageProvider _messenger;
-        private readonly ErrorHandler _errorHandler;
-
-        public BaseOAuthAuthentication(IMessageProvider messenger)
-        {
-            _messenger = messenger;
-            _errorHandler = new ErrorHandler(messenger);
-        }
-
-        public async Task Authenticate()
-        {
-            AuthenticationPrompt();
-        }
-
-        public async Task Expire()
-        {
-            ExpirePrompt();
-        }
+        public bool Authenticated { get; set; } = false;
+        public event EventHandler AuthenticationChanged;
 
-        public abstract void AuthenticationPrompt();
-        public abstract void ExpirePrompt();
+        protected HttpClient _client;
+        protected string _accessToken;
```

Comment:
Field, not private, are UpperCamelCase

Reply:
All fields use camelCase, public fields should not be used

Reply:
In line with MS design guidelines should use public nor protected. 

https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms229042(v=vs.100)?redirectedfrom=MSDN

> Do not provide instance fields that are public or protected.
> 
> Public and protected fields do not version well and are not protected by code access security demands. Instead of using publicly visible fields, use private fields and expose them through properties.